types/xdg_shell/wlr_xdg_surface.c

```diff
@@ -362,8 +366,12 @@ void handle_xdg_surface_commit(struct wlr_surface *wlr_surface) {
 	}
 
 	switch (surface->role) {
+	case WLR_XDG_SURFACE_ROLE_INERT_POPUP:
+		// inert popup
+		return;
 	case WLR_XDG_SURFACE_ROLE_NONE:
-		// inert toplevel or popup
+		// should not happen
+		wlr_log(WLR_ERROR, "xdg_surface %p commit with role == NONE", surface);
```

Comment:
should there be an assert here instead of an error?

Reply:
I wasn't sure if it should be an assert or a protocol error. I feel a buggy client could trigger this if committing without creating a role object? So in this case a protocol error might be better. What do you think?

Reply:
If a client manages to trigger that (and it is indeed forbidden by the protocol), then the client should be sent an error and disconnected.

If it is simply because of inert objects, there is no error at all.

Reply:
Changed this to an error, since inert popups are handled separately. I did not actually see any case that triggers this now; I might try to do a test with a client that intentionally tries to trigger this error.